src/core/radio/trel_interface.cpp

```diff
@@ -34,101 +34,421 @@
 
 #if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
 
-#include <openthread/platform/trel-udp6.h>
+#include <string.h>
 
+#include <openthread/platform/trel.h>
+
+#include "common/as_core_type.hpp"
 #include "common/code_utils.hpp"
+#include "common/debug.hpp"
 #include "common/instance.hpp"
 #include "common/locator_getters.hpp"
 #include "common/logging.hpp"
+#include "common/string.hpp"
+#include "net/dns_types.hpp"
+#include "utils/parse_cmdline.hpp"
 
 namespace ot {
 namespace Trel {
 
+const char Interface::kTxtRecordExtAddressKey[] = "mac";
+const char Interface::kTxtRecordExtPanIdKey[]   = "xpan";
```

Comment:
nit: to be consistent with meshcop service?
```suggestion
const char Interface::kTxtRecordExtAddressKey[] = "xa";
const char Interface::kTxtRecordExtPanIdKey[]   = "xp";
```

Reply:
I think `mac` and `xpan` are defined by spec?

Reply:
I think `mac` and `xpan` are defined by spec?

Reply:
TREL spec is still in draft and not final. Thread 1.1.1 already defines `xp` for Extended PAN ID.

Reply:
We can merge this PR as is, and later revise to `xa` and `xp`  after they are accepted by Spec. ?

Reply:
The purpose is to discuss whether or not to update the draft spec.

Reply:
+1 for using `xa` and `xp` for better consistency.

Reply:
I also prefer the smaller/shorter key names. Changed in new push.